modules/runner/src/testFixtures/java/org/apache/ignite/internal/ClusterConfiguration.java

```diff
@@ -0,0 +1,202 @@
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal;
+
+import java.nio.file.Path;
+import java.util.Objects;
+import org.junit.jupiter.api.TestInfo;
+
+/**
+ * Class for configuring {@link Cluster} instances.
+ */
+public class ClusterConfiguration {
+    public static final int DEFAULT_BASE_PORT = 3344;
+
+    public static final int DEFAULT_BASE_CLIENT_PORT = 10800;
+
+    public static final int DEFAULT_BASE_HTTP_PORT = 10300;
+
+    public static final int DEFAULT_BASE_HTTPS_PORT = 10400;
+
+    /** Default nodes bootstrap configuration pattern. */
+    private static final String DEFAULT_NODE_BOOTSTRAP_CFG = "ignite {\n"
+            + "  \"network\": {\n"
+            + "    \"port\":{},\n"
+            + "    \"nodeFinder\":{\n"
+            + "      \"netClusterNodes\": [ {} ]\n"
+            + "    }\n"
+            + "  },\n"
+            + "  clientConnector: { port:{} }\n"
+            + "  rest: {\n"
+            + "    port: {},\n"
+            + "    ssl.port: {}\n"
+            + "  }\n"
+            + "}";
+
+    private final TestInfo testInfo;
+
+    private final Path workDir;
+
+    private final String defaultNodeBootstrapConfigTemplate;
+
+    private final String clusterName;
+
+    private final int basePort;
+
+    private final int baseClientPort;
+
+    private final int baseHttpPort;
+
+    private final int baseHttpsPort;
+
+    private ClusterConfiguration(
+            TestInfo testInfo,
+            Path workDir,
+            String defaultNodeBootstrapConfigTemplate,
+            String clusterName,
+            int basePort,
+            int baseClientPort,
+            int baseHttpPort,
+            int baseHttpsPort
+    ) {
+        this.testInfo = testInfo;
+        this.workDir = workDir;
+        this.defaultNodeBootstrapConfigTemplate = defaultNodeBootstrapConfigTemplate;
+        this.clusterName = clusterName;
+        this.basePort = basePort;
+        this.baseClientPort = baseClientPort;
+        this.baseHttpPort = baseHttpPort;
+        this.baseHttpsPort = baseHttpsPort;
+    }
+
+    public TestInfo testInfo() {
+        return testInfo;
+    }
+
+    public Path workDir() {
+        return workDir;
+    }
+
+    public String defaultNodeBootstrapConfigTemplate() {
+        return defaultNodeBootstrapConfigTemplate;
+    }
+
+    public String clusterName() {
+        return clusterName;
+    }
+
+    public int basePort() {
+        return basePort;
+    }
+
+    public int baseClientPort() {
+        return baseClientPort;
+    }
+
+    public int baseHttpPort() {
+        return baseHttpPort;
+    }
+
+    public int baseHttpsPort() {
+        return baseHttpsPort;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Builder for {@link ClusterConfiguration}.
+     */
+    public static class Builder {
+        private TestInfo testInfo;
+
+        private Path workDir;
```

Comment:
How about making these final and initializing then in the builder's constructor? This will allow us to enforce their mandatory status statically

Reply:
I thought about that but looks like we never use this approach in other builders, so this will make this builder inconsistent with others. What do you think?

Reply:
I don't think this inconsistency is a problem; it seems more important to avoid error by making the user supply the mandatory paramters. Probably we should fix this in other builders as well